overlord/devicestate/devicemgr.go

```
@@ -599,6 +615,52 @@ func (m *DeviceManager) ensureOperational() error {
 	return nil
 }
 
+// maybeRestoreAfterReset attempts to restore the serial assertion with a
+// matching key in a post-factory reset scenario. It is possible that it is
+// called when the device was unregistered, but when doing so, the device key is
+// removed.
+func (m *DeviceManager) maybeRestoreAfterReset(device *auth.DeviceState) (keyID, serial string, err error) {
+	// there should be a serial assertion for the current model
+	serials, err := assertstate.DB(m.state).FindMany(asserts.SerialType, map[string]string{
+		"brand-id": device.Brand,
+		"model":    device.Model,
+	})
```

should we return an error if the err != nil and is not a not found error?